modules/nf-core/parabricks/fq2bam/main.nf

```diff
@@ -0,0 +1,66 @@
+process PARABRICKS_FQ2BAM {
+    tag "$meta.id"
+    label 'process_high'
+
+    // Exit if running this module with -profile conda / -profile mamba
+    if (workflow.profile.tokenize(',').intersect(['conda', 'mamba']).size() >= 1) {
+        exit 1, "Parabricks module does not support Conda. Please use Docker / Singularity / Podman instead."
+    }
+
+    container "nvcr.io/nvidia/clara/clara-parabricks:4.0.1-1"
+
+    input:
+    tuple val(meta), path(reads), path(interval_file)
+    tuple val(meta2), path(fasta)
+    tuple val(meta3), path(index)
+    path known_sites
+
+    output:
+    tuple val(meta), path("*.bam")                , emit: bam
+    tuple val(meta), path("*.bai")                , emit: bai
+    path "versions.yml"                           , emit: versions
+    path "qc_metrics", optional:true              , emit: qc_metrics
+    path("*.table"), optional:true                , emit: bqsr_table
+    path("duplicate-metrics.txt"), optional:true  , emit: duplicate_metrics
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def in_fq_command = meta.single_end ? "--in-se-fq $reads" : "--in-fq $reads"
+    def known_sites_command = known_sites ? known_sites.collect{"--knownSites $it"}.join(' ') : ""
+    def known_sites_output = known_sites ? "--out-recal-file ${prefix}.table" : ""
+    def interval_file_command = interval_file ? interval_file.collect{"--interval-file $it"}.join(' ') : ""
+    """
+
+    INDEX=`find -L ./ -name "*.amb" | sed 's/\\.amb\$//'`
+    # index and fasta need to be in the same dir as regular files (not symlinks)
+    #   and have the same base name for pbrun to function
+    #   here we copy the index into the staging dir of fasta
+    FASTA_PATH=`readlink -f $fasta`
+    cp \$INDEX.amb \$FASTA_PATH.amb
+    cp \$INDEX.ann \$FASTA_PATH.ann
+    cp \$INDEX.bwt \$FASTA_PATH.bwt
+    cp \$INDEX.pac \$FASTA_PATH.pac
+    cp \$INDEX.sa \$FASTA_PATH.sa
+
+    pbrun \\
+        fq2bam \\
+        --ref $fasta \\
+        $in_fq_command \\
+        --read-group-sm $meta.id \\
+        --out-bam ${prefix}.bam \\
+        $known_sites_command \\
+        $known_sites_output \\
+        $interval_file_command \\
+        --num-gpus $task.accelerator.request \\
+        $args
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+            pbrun: \$(echo \$(pbrun version 2>&1) | sed 's/^Please.* //' )
+    END_VERSIONS
+    """
```

Comment:
```suggestion
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def in_fq_command = meta.single_end ? "--in-se-fq $reads" : "--in-fq $reads"
    def known_sites_command = known_sites ? known_sites.collect{"--knownSites $it"}.join(' ') : ""
    def known_sites_output = known_sites ? "--out-recal-file ${prefix}.table" : ""
    def interval_file_command = interval_file ? interval_file.collect{"--interval-file $it"}.join(' ') : ""
    def metrics_command = args = "--out-duplicate-metrics duplicate-metrics.txt" ? "touch duplicate-metrics.txt" : ""
    """

    touch ${prefix}.bam
    touch ${prefix}.bam.bai
    $metrics_command
    """
```

I'd add something along that, to fake output when running with stub (cf https://www.nextflow.io/docs/edge/process.html#stub)

And add the ` -stub-run` at the end of each command in the test.yml file to create said files

Reply:
@maxulysse I did this in https://github.com/nf-core/modules/pull/2555/commits/59fee5ef827ed784d8512005bf46983bf432c63b
It unfortunately neuters the functionality of the tests, requires removing the GPU arguments from the docker config, etc. I moved the original test config and yml file to a backup location for now.

Reply:
True, but we can't have GPUs on GHA.
I think that way people would still be able to test locally without the stub option